java/memory/memory-core/src/main/java/org/apache/arrow/memory/BaseAllocator.java

```diff
@@ -915,6 +916,33 @@ public boolean add(final int nBytes) {
       return true;
     }
 
+    @Override
+    public boolean add(long nBytes) {
```

Comment:
I think we should cast int to long and call the other overload instead of duplicating the code

Reply:
Ah right, and upon removal we replace